packages/repository-tests/src/crud/transactions.suite.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  Entity,
+  IsolationLevel,
+  juggler,
+  model,
+  property,
+  Transaction,
+  TransactionalRepository,
+} from '@loopback/repository';
+import {expect, skipIf, toJSON} from '@loopback/testlab';
+import {Suite} from 'mocha';
+import {withCrudCtx} from '../helpers.repository-tests';
+import {
+  CrudFeatures,
+  CrudTestContext,
+  DataSourceOptions,
+  TransactionalRepositoryCtor,
+} from '../types.repository-tests';
+
+// Core scenarios for testing CRUD functionalities of Transactional connectors
+// Please keep this file short, put any advanced scenarios to other files
+export function transactionSuite(
+  dataSourceOptions: DataSourceOptions,
+  repositoryClass: TransactionalRepositoryCtor,
+  connectorFeatures: CrudFeatures,
+) {
+  skipIf<[(this: Suite) => void], void>(
+    !connectorFeatures.supportsTransactions,
+    describe,
+    `transactions`,
+    () => {
+      @model()
+      class Product extends Entity {
+        @property({
+          type: connectorFeatures.idType,
+          id: true,
+          generated: true,
+          description: 'The unique identifier for a product',
+        })
+        id: number | string;
+
+        @property({type: 'string', required: true})
+        name: string;
+
+        constructor(data?: Partial<Product>) {
+          super(data);
+        }
+      }
+
+      describe('create-retrieve with transactions', () => {
+        let repo: TransactionalRepository<Product, typeof Product.prototype.id>;
+        before(
+          withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+            repo = new repositoryClass(Product, ctx.dataSource);
+            await ctx.dataSource.automigrate(Product.name);
+          }),
+        );
+
+        it('retrieves model instance once transaction is committed', async () => {
+          const tx: juggler.Transaction = await repo.beginTransaction({
```

Comment:
We shouldn't be using `juggler` types here, let's use `Transaction` from `@loopback/repository` or don't specify the type at all.

```suggestion
          const tx = await repo.beginTransaction({
```

Same comment applies to all other tests too.